src/app/cache-data.ts

```diff
@@ -0,0 +1,10 @@
+import { GitHubOrganization } from "./git-hub-organization";
+
+export class CacheData {
+
+    public pageSize: number; 
+    public page: number;
+    public data: Promise<void | GitHubOrganization[]>;
```

Comment:
I made this simply a `GitHubOrganization[]` and cached the data itself rather than the Promise wrapping its data. But then I need to reconstruct the Promise when the cache is hit, so your approach is probably just as good.